### ISSUE TITLE
Email selected users when reimbursement requests are entered

### ISSUE THREAD

**Issue by volatile3** [rubyforgood/casa#4476]
## What type(s) of user does this feature affect?
 - supervisors
 - admins

## Description
See: https://github.com/rubyforgood/casa/issues/4474
For each new reimbursement request submitted, send an email to admins/supervisors who have reimbursement notifications enabled.

Email Subject: " New reimbursement request from [volunteer name]"
Email text: " [Volunteer name] has submitted a reimbursement request, please follow up on the reimbursements page
- Include a link to the reimbursements page

## Screenshots of current behavior, if any
![image](https://user-images.githubusercontent.com/123524778/214474229-ed165b5c-d07c-46fb-bc64-fa2abed994ee.png)

## QA Login Details
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 6-8pm Pacific), stakeholder news, and upcoming new issues.

**Comment by compwron** [comment]
We still want this!

**Comment by bacchist** [comment]
I'd like to work on this.

**Comment by compwron** [comment]
@bacchist go for it! Please reply here to confirm that you are working on it :)

**Comment by bacchist** [comment]
Yes, I am. :)

**Comment by bacchist** [comment]
I am still working on this, but I am also waiting for the option itself to be implemented as part of #4474

**Comment by littleforest** [comment]
@bacchist this looks unblocked now, are you still interested in working on it?

**Comment by bacchist** [comment]
@littleforest I'm sorry, I should have let somebody know that I was unable to work on this at the moment. I am still interested in it, but it will likely be about 2 or 3 weeks before I will have the free time to devote to it. I can certainly find another issue at that point, so consider it available if anyone else is interested.

**Comment by sarvaiyanidhi** [comment]
Hi @littleforest Can I take up this issue if it is still available?

**Comment by sarvaiyanidhi** [comment]
I am working on this issue. Will submit PR soon.

**Comment by sarvaiyanidhi** [comment]
Hello @littleforest .. I have submitted PR for this feature but wanted to check on couple of points
- If we submit case contact after selecting multiple cases in that case we will have multiple reimbursement requests and mail will sent for each reimbursement request of the case to the supervisor so should we also add in mail case number along with other details so supervisor can differentiate details accordingly.
- I also observed one more thing while testing the flow of reimbursement request
  - As a Volunteer I have created new case contact and want reimbursement for it.
  - Now supervisor reviews reimbursement and process it by marking Reimbursement complete
  - Later on, Volunteer goes back to case contact, edit it back and remove reimbursement which might be added wrongly or any other reason which may conflict if updated as reimbursement is already processed. 
  - I think if reimbursement is processed in that case Volunteer should not able to edit those details.

**Comment by sarvaiyanidhi** [comment]
Implementation is completed and have pushed PR #5208 for review.

**Comment by sarvaiyanidhi** [comment]
PR #208 is submitted for review.

**Comment by sarvaiyanidhi** [comment]
PR #5208 is under review for this issue.

**Comment by sarvaiyanidhi** [comment]
> Hello @littleforest .. I have submitted PR for this feature but wanted to check on couple of points
> 
> * If we submit case contact after selecting multiple cases in that case we will have multiple reimbursement requests and mail will sent for each reimbursement request of the case to the supervisor so should we also add in mail case number along with other details so supervisor can differentiate details accordingly.
> * I also observed one more thing while testing the flow of reimbursement request
>   
>   * As a Volunteer I have created new case contact and want reimbursement for it.
>   * Now supervisor reviews reimbursement and process it by marking Reimbursement complete
>   * Later on, Volunteer goes back to case contact, edit it back and remove reimbursement which might be added wrongly or any other reason which may conflict if updated as reimbursement is already processed.
>   * I think if reimbursement is processed in that case Volunteer should not able to edit those details.

@schoork I had also posted couple of queries on this issue. Please check if it is relevant to address, I can open new PR if we want the mentioned points to be handled in the code. Thanks

**Comment by schoork** [comment]
Good questions.

1. I don't think it creates multiple reimbursement requests, but I'm not 100% sure on that. The use case for multiple case contacts is when there are siblings and the volunteer is recording a touch point with all of them at the same time. In this case there should be one reimbursable expense, but based on this PR there are going to be multiple emails. I'd wait for the supervisors to come back and request these to be consolidated into one.
2. I agree that it should not be editable after it's approved. That's probably worth writing up as an issue. Just get Brandon's okay before working it.

**Comment by sarvaiyanidhi** [comment]
Thank you @schoork for your feedback. @bcastillo32 Can you please share your view on above point and if required please assign issue for me to work on it. Thanks
